### ISSUE TITLE
[no-unused-vars] False positive on module augmentation

### ISSUE THREAD

**Issue by sveyret** [typescript-eslint/typescript-eslint#2577]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "ignoreRestSiblings": true
      }
    ]
  }
}
```

```TS
// eslint-disable-next-line import/no-internal-modules
import type { Theme } from '@material-ui/core/styles/createMuiTheme'

declare module '@material-ui/core/styles/createMuiTheme' {
  interface Theme {
    application: {
      menuWidth: number
    }
  }
}
```

**Expected Result**

No error

**Actual Result**

2:10  error  'Theme' is defined but never used         @typescript-eslint/no-unused-vars

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.1` |
| `@typescript-eslint/parser`        | `4.1.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.9.0` |
| `node`                             | `14.4.0` |

**Comment by sveyret** [comment]
Note that the file is an ambient, trying to augment the `Theme` type from `material-ui`.

This was working with my previous version of eslint-plugin and parser (4.0.1).

**Comment by HolgerJeromin** [comment]
I have the same problem with a plain ts file

    module Hmi {
        export module System {

**Comment by HolgerJeromin** [comment]
related to #2573 ?

**Comment by bradzacher** [comment]
@HolgerJeromin - your problem is unrelated.

----

This is actually working as intended.
The `Theme` import is actually unused by TS for the module augmentation.

You only need an import statement with no specifiers.

```ts
import '@material-ui/core/styles/createMuiTheme'

declare module '@material-ui/core/styles/createMuiTheme' {
  interface Theme {
    application: {
      menuWidth: number
    }
  }
}
```

Why is the import statement required at all? Because it tells typescript "this is **not** a declaration of a **new** module, it's an **augmentation** of an **existing** module".

Here is an example in our codebase:
https://github.com/typescript-eslint/typescript-eslint/blob/13292942d063c3ab307197b0dbe78cee3a0d9743/packages/eslint-plugin/typings/typescript.d.ts

If I change the import in the above file from
`import 'typescript';`
to
`import {TypeChecker, Type} from 'typescript';`

This is what happens using TS's built in `noUnusedLocals` compiler option:
![image](https://user-images.githubusercontent.com/7462525/93623766-15890a00-f994-11ea-9b11-983cfd52b12b.png)
